devops/jobs/ConfigurationWatcher.groovy

```diff
@@ -0,0 +1,154 @@
+            }
+            logRotator common_logrotator
+
+            def access_control = extraVars.get('ACCESS_CONTROL',[])
+            access_control.each { acl ->
+                common_read_permissions.each { perm ->
+                    authorization {
+                        permission(perm,acl)
+                    }
+                }
+            }
+
+            // for GitHub Webhooks
+            properties {
+                githubProjectUrl(extraVars.get("EDX_REPO_ROOT") +  "configuration/")
+            }
+
+            scm {
+                // check out edx/configuration repository from GitHub
+                git {
+                    remote {
+                        url(extraVars.get("CONFIGURATION_REPO_URL"))
+                        branch(config_branch)
+                    }
+                    extensions {
+                        relativeTargetDirectory('configuration')
+                    }
+                }
+                
+            }
+
+            triggers {
+                // due to a bug or misconfiguration, jobs with default branches with
+                // slashes are indiscriminately triggered by pushes to other branches.
+                // For more information, see:
+                // https://openedx.atlassian.net/browse/TE-1921
+                // for commits merging into master, trigger jobs via github pushes
+                    if (config_branch == 'master') {
+                        githubPush()
+                    }
+                // for all other jobs in this style, poll github for new commits on
+                // the 'defaultBranch' every 10 minutes
+                    else {
+                        scm('H/10 * * * *')
+                    }
+            }
+            
+            // run the trigger-builds shell script in a virtual environment called venv
+            steps {
+                virtualenv {
+                    name('venv')
+                    nature('shell')
+                    command dslFactory.readFileFromWorkspace('devops/resources/trigger-builds.sh')
+                }
+
+                 // inject environment variables defined in the temp_props file (TO_BUILD)
+                environmentVariables {
+                    propertiesFile('temp_props')
+                }
+
+                conditionalSteps {
+                    // ensure TO_BUILD is not empty string to avoid failed build if nothing to be built
+                    condition {
+                        not {
+                            stringsMatch('', '${TO_BUILD}', false)
+                        }
+                    }
+
+                    steps {
+
+                        // trigger the jobs defined in the TO_BUILD environment variable; this is set via the trigger-builds script
+                        // and injected into the environment from the temp_props file
+                        downstreamParameterized {
+                            trigger('${TO_BUILD}') 
```

Comment:
Ah, yes, I remember this now.  I think this is probably ok in retrospect, I'm not sure how to make it clearer honestly.